hangman.rb

```diff
@@ -0,0 +1,184 @@
+require "Colorize"
+
+class Hangman
+
+  def initialize
+
+    #this runs at the beginning and calls lots of other methods
+
+    # initialized variables
+    @answer = pick_answer
+    # User has 7 wrong guesses before game over
+    @no_of_guesses = 0
+    @blanks_array = answer_to_blanks
+    @guessed_letters = [ ]
+
+    # methods we need to call in the flow of the game
+    start
+    art
+    get_guess
+
+  end
+
+  def start
+    puts "Welcome to Hangman! Type " + "'quit'".colorize(:red) + " at any time to exit."
+  end
+
+
+# creates an answer from the array that is randomized
+  def pick_answer
+    # randomly pick the answer from this array
+    library = ["cat", "dog", "mouse", "letter", "elephant",
+      "bottomless", "house", "polecat", "coffee", "apple",
+      "laptop", "caramel", "sushi", "computer", "television"]
+    @answer = library.sample
+    return @answer
+  end
+
+  def answer_to_blanks
+    # convert the @answer to same number of blanks
+    # so we can display that clue for the user
+    num_of_blanks = @answer.length
+    @blanks = "_" * num_of_blanks
+    puts @blanks
+
+    # split up the blank array into individual "_"  strings
+    # so we can change each one individually
+    @blanks_array = @blanks.split(//)
+    return @blanks_array
+
+  end
+
+
+  def get_guess
+    # gets user input
+    puts "\nGuess a letter: "
+    @guess = gets.chomp
+
+    if @guess.upcase == "QUIT"
+      exit
+    end
+
+    valid_guess
+    check_answer
+  end
+
+  # validates the guess to make sure it's a through z
+  def valid_guess
+    until  (("A".."Z").include? @guess) || (("a".."z").include? @guess)
+      puts "Please enter a letter from A to Z."
+      @guess = gets.chomp
+    end
+  end
+
+
+  def check_answer
+    # compares guess to the answer
+
+    # answer letters go into an array
+    answer_array = @answer.split(//)
+    # checks the guess against each letter (including duplicates)
+    answer_array.each_index do |index|
```

Comment:
You can actually use the enumerable method `each_with_index` here so you wouldn't need to use `answer_array[index]` since you would have the element as well as the index